### DISCUSSION TITLE
Can't get it to work

### DISCUSSION THREAD

**Discussion by iBlueDragon** [kingsleyadam/local-abbfreeathome-hass#116] [Q&A] [answered]
Hi,

I don't know what I am doing wrong, but I can't get the integration to work. I always get a setup error.

SysAP version: 3.4.1
API is enabled. SysAP gets discovered and I am asked for username and password. I tried the installer user and other users I set up.

This is the log:
Logger: homeassistant.config_entries
Quelle: config_entries.py:640
Erstmals aufgetreten: 19:03:54 (1 Vorkommnisse)
Zuletzt protokolliert: 19:03:54

Error setting up entry free@home System Access Point (ABB7F59D6A11) for abbfreeathome_ci
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 640, in __async_setup_with_context
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/abbfreeathome_ci/__init__.py", line 109, in async_setup_entry
    await _free_at_home.load_devices()
  File "/usr/local/lib/python3.13/site-packages/abbfreeathome/freeathome.py", line 139, in load_devices
    await self._load_devices_by_function(_function, _device_class)
  File "/usr/local/lib/python3.13/site-packages/abbfreeathome/freeathome.py", line 154, in _load_devices_by_function
    device_class(
    ~~~~~~~~~~~~^
        device_id=_device.get("device_id"),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<8 lines>...
        room_name=_device.get("room_name"),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/usr/local/lib/python3.13/site-packages/abbfreeathome/devices/cover_actuator.py", line 61, in __init__
    super().__init__(
    ~~~~~~~~~~~~~~~~^
        device_id,
        ^^^^^^^^^^
    ...<8 lines>...
        room_name,
        ^^^^^^^^^^
    )
    ^
  File "/usr/local/lib/python3.13/site-packages/abbfreeathome/devices/base.py", line 46, in __init__
    self._refresh_state_from_outputs()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/usr/local/lib/python3.13/site-packages/abbfreeathome/devices/base.py", line 151, in _refresh_state_from_outputs
    self._refresh_state_from_output(_output)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/usr/local/lib/python3.13/site-packages/abbfreeathome/devices/cover_actuator.py", line 161, in _refresh_state_from_output
    self._tilt_position = int(output.get("value"))
                          ~~~^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: '0.392157'

Your help is highly appreciated!

iBlueDragon

**Comment by iBlueDragon** [comment, marked as answer]
I tried again this morning and it worked after entering username and PW again. Didn't change anything...

Thanks for the good work!
iBlueDragon

**Comment by kingsleyadam** [comment]
A bit strainge, it looks like the tilt position came through as `0.392157` and Python couldn’t convert it to an integer type.  You could download the diagnostics for the integration to see what the current value is for this. It’s already scrubbed data so it’s safe to paste here if you want.

Info on how to do this is here: https://github.com/kingsleyadam/local-abbfreeathome-hass?tab=readme-ov-file#download-diagnostics

**Comment by kingsleyadam** [comment]
Just as a note, this was addressed in a recent release. We’re handling floats and converting them to integers instead of attempting to convert a string directly to an interger.

**Reply by iBlueDragon** [reply]
Hi Adam, thanks for the update. After Christmas I had to reinstall because of a network change and ran into the same issue. I moved all blinds to the up position and the setup went through. Good to know it's fixed in the integration now.
